cy_ioctl.go

```diff
@@ -673,15 +668,14 @@ var (
 // is sdpcm_process_rx_packet in reference.
 //
 //	reference: sdpcm_process_rx_packet
-func (d *Device) sdpcmProcessRxPacket(buf []byte) (payloadOffset, plen uint32, flag whd.SDPCMHeaderType, err error) {
-	const badFlag = 0
-	const sdpcmOffset = 0
-	hdr := whd.DecodeSDPCMHeader(buf[sdpcmOffset:])
+func (d *Device) sdpcmProcessRxPacket(buf []byte) (payloadOffset, plen uint32, header whd.SDPCMHeaderType, err error) {
+	const badHeader = whd.UNKNOWN_HEADER
+	hdr := whd.DecodeSDPCMHeader(buf)
 	switch {
 	case hdr.Size != ^hdr.SizeCom&0xffff:
```

Comment:
Gonna make a mental note here since I think this was me: avoid ambiguous looking bit operations, use parenthesis to group operations when order not immediately obvious.

Reply:
I'm not following your comment on this one about bit operations.

I saw:

```
sdpcmOffset := 0
buf[sdpcmOffset:]
```

As:

`buf[0:]`

Which is just:

`buf`

???  Maybe I missed something subtle.

Reply:
Oh no- the issue is `^hdr.SizeCom&0xffff`, which begs the question if the not is applied before or after the and. In this case it seems to be before but that should be explicit.

Reply:
Oh ya, I can't guess the right answer...need parens